source/extensions/omni.isaac.lab_tasks/test/test_environments.py

```diff
@@ -80,16 +84,16 @@ def test_single_instance_gpu(self):
     """
 
     def _check_random_actions(self, task_name: str, device: str, num_envs: int, num_steps: int = 1000):
-        """Run random actions and check environments return valid signals."""
+        """Run random actions and check environments returned signals are valid."""
         # create a new stage
         omni.usd.get_context().new_stage()
         # parse configuration
         env_cfg: ManagerBasedRLEnvCfg = parse_env_cfg(task_name, device=device, num_envs=num_envs)
         # create environment
         env: ManagerBasedRLEnv = gym.make(task_name, cfg=env_cfg)
-        # this flag is necessary to prevent a bug where the simulation gets stuck randomly when running the
-        # test on many environments.
-        env.sim.set_setting("/physics/cooking/ujitsoCollisionCooking", False)
+
+        # disable control on stop
+        env.unwrapped.sim._app_control_on_stop_handle = None  # type: ignore
```

Comment:
Curious: what does this do?

Reply:
By default the simulation closes itself if it "stops" playing. This line here disables this behavior so we can have multiple checks without shutting down the simulation app.